x-pack/solutions/observability/plugins/observability_logs_explorer/public/logs_explorer_url_schema.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TimeRange, RefreshInterval, Query } from '@kbn/data-plugin/common/types';
+import { DiscoverAppState } from '@kbn/discover-plugin/public';
+import { SMART_FALLBACK_FIELDS } from '@kbn/discover-utils';
+import { ExistsFilter, Filter, PhrasesFilter } from '@kbn/es-query';
+import { FILTERS } from '@kbn/es-query';
+import { PhraseFilterValue } from '@kbn/es-query/src/filters/build_filters';
+
+import {
+  Column,
+  DataSourceSelectionPlain,
+  DataViewSelectionPayload,
+  SingleDatasetSelectionPayload,
+  UnresolvedDatasetSelectionPayload,
+  DataViewSpecWithId,
+  UrlSchemaV1,
+  UrlSchemaV2,
+  ControlsState,
+  DisplayOptions,
+  OptionsListControl,
+  ControlOptions,
+} from './logs_explorer_schema_types';
+
+export interface LogsExplorerPublicState {
+  chart?: {
+    breakdownField?: string | null;
+  };
+  controls?: ControlsState;
+  dataSourceSelection?: DataSourceSelectionPlain;
+  filters?: Filter[];
+  grid?: {
+    columns?: Column[];
+    rows?: {
+      rowHeight?: number;
+      rowsPerPage?: number;
+    };
+  };
+  query?: Query;
+  refreshInterval?: RefreshInterval;
+  time?: TimeRange;
+}
+
+export const hydrateDataSourceSelection = (
+  dataSourceSelection: DataSourceSelectionPlain
+): DataViewSpecWithId => {
+  if (dataSourceSelection.selectionType === 'all') {
+    return {
+      id: 'discover-observability-solution-all-logs',
```

Comment:
How about we use the constant `ALL_LOGS_DATA_VIEW_ID` here to make sure it stays in sync?